src/test/java/jenkins/plugins/http_request/HttpRequestStepTest.java

```diff
@@ -0,0 +1,499 @@
+            "    consoleLogResponseBody: true,\n" +
+            "    httpMode: '"+mode.toString()+"'\n" +
+            "println('Status: '+response.getStatus())\n" +
+            "println('Response: '+response.getContent())\n",
+            true));
+
+        // Execute the build
+        WorkflowRun run = proj.scheduleBuild2(0).get();
+
+        // Check expectations
+        j.assertBuildStatusSuccess(run);
+
+        if (mode == HttpMode.HEAD) return;
+
+        j.assertLogContains(allIsWellMessage,run);
+    }
+
+    @Test
+    public void invalidResponseCodeFailsTheBuild() throws Exception {
+        // Prepare the server
+        final HttpHost target = start();
+        final String baseURL = "http://localhost:" + target.getPort();
+
+        // Configure the build
+        WorkflowJob proj = j.jenkins.createProject(WorkflowJob.class, "proj");
+        proj.setDefinition(new CpsFlowDefinition(
+            "def response = httpRequest url:'"+baseURL+"/invalidStatusCode',\n" +
+            "    consoleLogResponseBody: true\n" +
+            "println('Status: '+response.getStatus())\n" +
+            "println('Response: '+response.getContent())\n",
+            true));
+
+        // Execute the build
+        WorkflowRun run = proj.scheduleBuild2(0).get();
+
+        // Check expectations
+        j.assertBuildStatus(Result.FAILURE, run);
+        j.assertLogContains("Throwing status 400 for test",run);
+    }
+
+    @Test
+    public void invalidResponseCodeIsAccepted() throws Exception {
+        // Prepare the server
+        final HttpHost target = start();
+        final String baseURL = "http://localhost:" + target.getPort();
+
+        // Configure the build
+        WorkflowJob proj = j.jenkins.createProject(WorkflowJob.class, "proj");
+        proj.setDefinition(new CpsFlowDefinition(
+            "def response = httpRequest url:'"+baseURL+"/invalidStatusCode',\n" +
+            "    consoleLogResponseBody: true,\n" +
+            "    validResponseCodes: '100:599'\n" +
+            "println('Status: '+response.getStatus())\n" +
+            "println('Response: '+response.getContent())\n",
+            true));
+
+        // Execute the build
+        WorkflowRun run = proj.scheduleBuild2(0).get();
+
+        // Check expectations
+        j.assertBuildStatusSuccess(run);
+        j.assertLogContains("Throwing status 400 for test",run);
+    }
+
+    @Test
+    public void sendAllContentTypes() throws Exception {
+        for (MimeType mimeType : MimeType.values()) {
+            sendContentType(mimeType);
+        }
+    }
+
+    public void sendContentType(final MimeType mimeType) throws Exception {
+        // Prepare the server
+        final HttpHost target = start();
+        final String baseURL = "http://localhost:" + target.getPort();
+
+        // Configure the build
+        WorkflowJob proj = j.jenkins.createProject(WorkflowJob.class, "proj"+mimeType.toString());
```

Comment:
Hmm, this ought to fail due to the `/`. Seems that [`createProject`](https://github.com/jenkinsci/jenkins/blob/5627b7b456b24aca96f672c887bbf6c643415040/core/src/main/java/hudson/model/ItemGroupMixIn.java#L310) is calling `ProjectNamingStrategy` but neglecting to call [`checkGoodName`](https://github.com/jenkinsci/jenkins/blob/5627b7b456b24aca96f672c887bbf6c643415040/core/src/main/java/jenkins/model/Jenkins.java#L3133-L3165). Cf. https://github.com/jenkinsci/cloudbees-folder-plugin/pull/3. In the meantime, API clients should take care to only supply legitimate names.